modules/xmpp/Polls.ts

```diff
@@ -0,0 +1,110 @@
+import { XMPPEvents } from '../../service/xmpp/XMPPEvents';
+
+import ChatRoom from './ChatRoom';
+import XMPP from './xmpp';
+
+export const COMMAND_ANSWER_POLL = 'answer-poll';
+export const COMMAND_NEW_POLL = 'new-poll';
+export const COMMAND_OLD_POLLS = 'old-polls';
+
+export default class Polls {
+    private _mainRoom: ChatRoom;
+    private _xmpp: XMPP;
+
+    /**
+     * Constructs polls for a room.
+     *
+     * @param {ChatRoom} room the main room.
+     */
+    constructor(room: ChatRoom) {
+        this._mainRoom = room;
+        this._xmpp = room.xmpp;
+
+        this._handleMessages = this._handleMessages.bind(this);
+        this._mainRoom.xmpp.addListener(XMPPEvents.POLLS_EVENT, this._handleMessages);
+    }
+
+    /**
+     * Whether polls is supported on backend.
+     *
+     * @returns {boolean} whether polls is supported on backend.
+     */
+    isSupported() {
+        return Boolean(this._xmpp.pollsComponentAddress);
+    }
+
+    /**
+     * Stops listening for events.
+     */
+    dispose() {
+        this._mainRoom.xmpp.removeListener(XMPPEvents.POLLS_EVENT, this._handleMessages);
+    }
+
+    /**
+     * Creates and sends a new poll.
+     *
+     * @param pollId
+     * @param question
+     * @param answers
+     */
+    createPoll(pollId: string, question: string, answers: Array<{ name: string; }>) {
+        this._mainRoom.sendPrivateMessage(
+            this._xmpp.pollsComponentAddress,
+            JSON.stringify({
+                answers: answers.map(answer => answer.name),
+                pollId,
+                question,
+                roomJid: this._mainRoom.roomjid,
+                type: COMMAND_NEW_POLL
+            }),
+            'json-message',
+            true);
+    }
+
+    /**
+     * Sends answers for a poll.
+     *
+     * @param pollId
+     * @param answers
+     */
+    answerPoll(pollId: string, answers: Array<boolean>) {
+        this._mainRoom.sendPrivateMessage(
+            this._xmpp.pollsComponentAddress,
+            JSON.stringify({
+                answers,
+                pollId,
+                type: COMMAND_ANSWER_POLL
+            }),
+            'json-message',
+            true);
+    }
+
+    /**
+     * Handles a message for polls.
+     *
+     * @param {object} payload - Arbitrary data.
+     */
+    _handleMessages(payload) {
+        switch (payload.type) {
+        case COMMAND_NEW_POLL:
+            this._mainRoom.eventEmitter.emit(XMPPEvents.POLLS_RECEIVE_EVENT, payload);
+
+            break;
+        case COMMAND_OLD_POLLS: {
+            if (payload?.polls) {
+                payload.polls.forEach((poll: any) => {
```

Comment:
Define a type for Poll?

Reply:
It will be best to move the types from jitsi-meet to ljm and use those in jitsi-meet. I will add a todo there, once we can use them that way, I will move them.